jme3-core/src/main/java/com/jme3/texture/FrameBuffer.java

```diff
@@ -260,12 +259,35 @@ public void addColorTarget(FrameBufferBufferTarget colorBuf){
         colorBuf.slot=colorBufs.size();
         colorBufs.add(colorBuf);
     }
-
+    
     public void addColorTarget(FrameBufferTextureTarget colorBuf){
         // checkSetTexture(colorBuf.getTexture(), false);  // TODO: this won't work for levels.
         colorBuf.slot=colorBufs.size();
         colorBufs.add(colorBuf);
     }
+    
+    /**
+     * Sets the color target at the index.
+     * 
+     * @param i
+     * @param colorBuf 
+     */
+    public void setColorTarget(int i, FrameBufferTextureTarget colorBuf) {
+        colorBuf.slot = i;
+        colorBufs.set(i, colorBuf);
+    }
+    
```

Comment:
In case it is an ilegal state replace might be a better name. I would be fine with set too if it is clearly documented and throws a meaningful exception